uxarray/io/_ugrid.py

```diff
@@ -93,30 +93,42 @@ def _encode_ugrid(ds):
     """Encodes an unstructured grid represented under a ``Grid`` object as a
     ``xr.Dataset`` with an updated grid topology variable."""
 
-    if "grid_topology" in ds:
-        ds = ds.drop_vars(["grid_topology"])
+    # Create a copy of the input dataset to avoid modifying the original Grid._ds
+    out_ds = ds.copy()
 
-    grid_topology = ugrid.BASE_GRID_TOPOLOGY_ATTRS
+    if "grid_topology" in out_ds:
+        out_ds = out_ds.drop_vars(["grid_topology"])
 
-    if "n_edge" in ds.dims:
-        grid_topology["edge_dimension"] = "n_edge"
+    grid_topology_attrs = ugrid.BASE_GRID_TOPOLOGY_ATTRS.copy()  # Use a copy
```

Comment:
No need for `copy()` since none of the keys below are present in `ugrid.BASE_GRID_TOPOLOGY_ATTRS`, but nice to have to stay 100% safe in the future (even the the logic here seems like get the ugrid base attributes from dictionary first and add more not modify the base).

Reply:
Thanks for this, totally makes sense. I have made the change.